src/vm/siginfo.cpp

```diff
@@ -4987,16 +5030,21 @@ void ReportPointersFromValueType(promote_func *fn, ScanContext *sc, PTR_MethodTa
 void ReportPointersFromValueTypeArg(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, ArgDestination *pSrc)
 {
     WRAPPER_NO_CONTRACT;
-    
-    if (!pMT->ContainsPointers())
-        return;
+
+    if(pMT->ContainsPointers())
```

Comment:
(I would be useful to add some comments if it needs to be what you have written.)